src/Psy/Plugin/AbstractPlugin.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Psy\Plugin;
+
+abstract class AbstractPlugin
+{
+    public static function register()
+    {
+        Manager::register(new static(), static::getName());
+    }
+
+    /**
+     * @return string
+     */
+    final public static function getName()
+    {
+        $class = new \ReflectionClass(get_called_class());
+
+        return preg_replace('#Plugin$#', '', $class->getShortName());
```

Comment:
Add an abstract static function that plugins have to implement and call it a day. That way plugins can just use their Packagist package names. Mine can be called `bobthecow/psysh-devtools` and yours can be called `markcial/psysh-devtools`, and we'll never collide :)